src/tiledb/cloud/dag/dag.py

```diff
@@ -103,32 +127,12 @@ def __init__(
         _download_results: Optional[bool] = None,
         _internal_prewrapped_func: Callable[..., "results.Result[_T]"] = None,
         _internal_accepts_stored_params: bool = True,
-        **kwargs,
-    ):
-        """
-        Node is a class that represents a function to run in a DAG
-        :param func: function to run
-        :param args: tuple of arguments to run
-        :param name: optional name of dag
-        :param dag: dag this node is associated with
-        :param mode: Mode the node is to run in.
-        :param _download_results: An optional boolean to override default
-            result-downloading behavior. If True, will always download the
-            results of the function immediately upon completion.
-            If False, will not download the results of the function immediately,
-            but will be downloaded when ``.result()`` is called.
-        :param _prewrapped_func: For internal use only. A function that returns
-            something that is already a Result, which does not require wrapping.
-            We assume that all prewrapped functions make server calls.
-        :param _accepts_stored_params: For internal use only.
-            Applies only when ``_prewrapped_func`` is used.
-            ``True`` if ``_prewrapped_func`` can accept stored parameters.
-            ``False`` if it cannot, and all parameters must be serialized.
-        :param kwargs: dictionary for keyword arguments
-        """
+        **kwargs: Any,
+    ) -> None:
         self.id = uuid.uuid4()
+        """UUID for Node instance."""
```

Comment:
@spencerseale what would you think about using Python comments instead of unassigned strings for this PR?

```suggestion
        # UUID for Node instance.
```

I actually think most of the comments you've added are pretty clear from the code and can be omitted. They won't surface in documentation.

Reply:
These strings are just meant to materialize the attribute definitions in the docs rather than for someone looking through the code. I definitely agree on not overdoing the comments! My main intent was to get users all information about these classes into the api ref, where currently there is nothing: https://tiledb-inc.github.io/TileDB-Cloud-Py/reference/dag.dag.html

Reply:
@spencerseale Great! I had forgotten that attribute docstrings are a thing. In https://discuss.python.org/t/revisiting-attribute-docstrings/36413 I see that they're pretty widely used and appreciated. I'll amend my review.